## Final Project Submission

Please fill out:
* **Student name:**                       Kinyanjui Chris Kamau
* **Student pace:**                       full time
* **Scheduled project review date/time:** 
* **Instructor name:**                    Antonny Muiko



# Data Understanding.

In [15]:
#importing the relevant modules

#pandas
import pandas as pd

# matlab
import matplotlib.pyplot as plt

# to ensure all data visualizations appear on the same notebook and also stored here.
%matplotlib inline 

#seaborn
import seaborn as sns

#sqlite3
import sqlite3

#csv
import csv

#regex
import re

First file I shall load then explore is the ```'tn.movie_budgets.csv.gz'```.

In [2]:
# using pandas to load the file into movie_budget_db

movie_budget_df = pd.read_csv('zippedData/tn.movie_budgets.csv.gz')

movie_budget_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [21]:
#let us define some functions to get relevant information from the Data Frames.

#get information

def get_info(data):
    info_ = data.info()
    return info_
#get column names

def get_columns(data):
    columns=data.columns
    return columns

#type casting function for any column of numerical data stored as a string.
def int_conv(data,col_name):
    #first we remove the $ sign for the typecasting to work.
    data[col_name] = data[col_name].map(lambda col: re.sub('[^0-9]','', col))
    
    #then we convert/typecast.
    data[col_name] = data[col_name].astype(int)
    return data[col_name]


In [19]:
# getting info for the DataFrame "movie_budget_db"
get_info(movie_budget_df)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


# Analysis.

The DataFrame has a total of 6 columns and 5782 rows.

From the information above it is clear it has no missing value.

On further investigation some data types are in string form which will make 
it hard to sort them and use them for statistical analysis.

The problem columns are ```'production_budget'```,```'domestic_gross'``` and ```'worldwide_gross'```.

In [22]:
#type casting the three columns
#production_budget
int_conv(movie_budget_df,'production_budget')

0       425000000
1       410600000
2       350000000
3       330600000
4       317000000
          ...    
5777         7000
5778         6000
5779         5000
5780         1400
5781         1100
Name: production_budget, Length: 5782, dtype: int64

In [23]:
#domestic_gross
int_conv(movie_budget_df,'domestic_gross')


0       760507625
1       241063875
2        42762350
3       459005868
4       620181382
          ...    
5777            0
5778        48482
5779         1338
5780            0
5781       181041
Name: domestic_gross, Length: 5782, dtype: int64

In [24]:
#worldwide_gross
int_conv(movie_budget_df,'worldwide_gross')

0       2776345279
1       1045663875
2        149762350
3       1403013963
4       1316721747
           ...    
5777             0
5778        240495
5779          1338
5780             0
5781        181041
Name: worldwide_gross, Length: 5782, dtype: int64

In [25]:
get_info(movie_budget_df)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   int64 
 4   domestic_gross     5782 non-null   int64 
 5   worldwide_gross    5782 non-null   int64 
dtypes: int64(4), object(2)
memory usage: 271.2+ KB


In [26]:
# defining a function to look for duplicates from the DataFrames.
def get_duplicate(data):
    duplicate = data.duplicated().value_counts()
    return duplicate

In [27]:
get_duplicate(movie_budget_df)

False    5782
dtype: int64

From the info above the Data Frame ```'movie_budget_df'``` doesnt have any duplicate rows.

In [ ]:
review_df = pd.read_csv('zippedData/rt.reviews.tsv.gz', delimiter = '\t', encoding = 'latin-1')
review_df.head()

In [ ]:
gross_df = pd.read_csv('zippedData/bom.movie_gross.csv.gz', encoding = 'latin-1')
gross_df.head()

In [ ]:
imdb_df = pd.read_csv('zippedData/tmdb.movies.csv.gz',delimiter= "," , encoding = 'latin-1')
imdb_df.head()

In [ ]:
b = pd.read_csv('zippedData/tn.movie_budgets.csv.gz', encoding = 'latin-1')
b.head()

In [ ]:
import sqlite3

conn = sqlite3.connect('zippedData/im.db')
pd.read_sql('''
SELECT *
FROM movie_basics
''',conn)